### PR TITLE
fix: Enable to use size prop in SegmentedControl.Button component

### PR DESCRIPTION
**Pull request by RayRedGoose**
## Summary

Fixes: #1756  

`SegmentedControl.Button` can have `size` prop to modify button and and icon size with a `small`, `medium` or `large` value.

category

### CODE REVIEW

File: modules/react/segmented-control/lib/SegmentedControlButton.tsx [hunk 1]
```diff
@@ -79,13 +94,17 @@ const StyledButton = styled(BaseButton)<StyledType>(
...
       },
     },
+  }),
+  ({size}: ButtonContainerProps) => ({
+    ...getSizeStyles(size),
   })
 );
 
-export interface SegmentedControlButtonProps {
+export interface SegmentedControlButtonProps extends ButtonContainerProps {
```
Comment (mannycarrera4) [inline modules/react/segmented-control/lib/SegmentedControlButton.tsx:103]: yassssss

File: modules/react/segmented-control/lib/SegmentedControlButton.tsx [hunk 2]
```diff
@@ -35,12 +40,22 @@ const getIconButtonColors = (toggled?: boolean): ButtonColors => {
   };
 };
 
-const StyledButton = styled(BaseButton)<StyledType>(
+const getSizeStyles = (size?: ButtonSizes) => {
```
Comment (mannycarrera4) [inline modules/react/segmented-control/lib/SegmentedControlButton.tsx:43]: don't we also have `extraSmall` ?

**Review by mannycarrera4 — APPROVED** [review]
thank you for fixing this!

File: modules/react/segmented-control/lib/SegmentedControlButton.tsx (hunk 2 above)
Comment (RayRedGoose) [reply modules/react/segmented-control/lib/SegmentedControlButton.tsx:43]: no, SegmentedControl has only 3 size

**Pull request merged**